src/server_manager/web_app/digitalocean_app/connect_account_app.ts

```diff
@@ -0,0 +1,164 @@
+  @property({type: Object}) notificationManager: OutlineNotificationManager = null;
+
+  private session: OauthSession;
+
+  static get styles() {
+    return [
+      COMMON_STYLES, css`
+      :host {
+      }
+      .container {
+        display: flex;
+        flex-direction: column;
+        justify-content: center;
+        height: 100%;
+        align-items: center;
+        padding: 132px 0;
+        font-size: 14px;
+      }
+      #connectAccount img {
+        width: 48px;
+        height: 48px;
+        margin-bottom: 12px;
+      }
+      .card {
+        display: flex;
+        flex-direction: column;
+        align-items: stretch;
+        justify-content: space-between;
+        margin: 24px 0;
+        padding: 24px;
+        background: var(--background-contrast-color);
+        box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.14), 0 2px 2px 0 rgba(0, 0, 0, 0.12), 0 1px 3px 0 rgba(0, 0, 0, 0.2);
+        border-radius: 2px;
+      }
+      @media (min-width: 1025px) {
+        paper-card {
+          /* Set min with for the paper-card to grow responsively. */
+          min-width: 600px;
+        }
+      }
+      .card p {
+        color: var(--light-gray);
+        width: 100%;
+        text-align: center;
+      }
+      .card paper-button {
+        color: var(--light-gray);
+        width: 100%;
+        border: 1px solid rgba(255, 255, 255, 0.12);
+        border-radius: 2px;
+      }
+      .card paper-button[disabled] {
+        color: var(--medium-gray);
+        background: transparent;
+      }
+      /* Mirror images */
+      :host(:dir(rtl)) .mirror {
+        transform: scaleX(-1);
+      }
+    `
+    ];
+    // TODO: RTL
+  }
+
+  render() {
+    return html`
+    <outline-step-view id="connectAccount">
+      <span slot="step-title">${this.localize('oauth-connect-title')}</span>
+      <span slot="step-description">${this.localize('oauth-connect-description')}</span>
+      <paper-card class="card">
+        <div class="container">
+          <img src="images/digital_ocean_logo.svg">
+          <p>${this.localize('oauth-connect-tag')}</p>
+        </div>
+        <paper-button @tap="${this.onCancel}">${this.localize('cancel')}</paper-button>
+      </paper-card>
+    </outline-step-view>`;
+  }
+
+  async start(): Promise<DigitalOceanAccount> {
+    this.session = runDigitalOceanOauth();
+
+    let accessToken;
+    try {
+      accessToken = await this.session.result;
+    } catch (error) {
+      if (this.session.isCancelled()) {
+        this.onCancel();
+      } else {
+        console.error(`DigitalOcean authentication failed: ${error}`);
+        this.notificationManager.showError(this.localize('error-do-auth'));
+        throw error;
+      }
+    }
+
+    const data = await this.createAccountData(accessToken);
+    return this.createAccountModel(data);
+  }
+
+  async createAccountModel(data: account.Data): Promise<DigitalOceanAccount> {
+    const accessToken = data.credential as string;
```

Comment:
Here we could have something like `account = digitalOceanCloud.connectAccount(accessToken)`

digitalOceanCloud should encapsulate sentry, storage, debug mode, image, etc. That's probably the only thing you need to inject into this app.